### DISCUSSION TITLE
[help] How can I tell the reason that a target is outdated?

### DISCUSSION THREAD

**Discussion by billdenney** [ropensci/targets#1052] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm working to update the nlmixr2targets package.  For that, some targets get run when I do not think that they should be run.  When I run `tar_outdated()`, I see that the targets are outdated, but I don't see the reason a target is outdated.

My guess is that the `digest()` of `nlmixr2` object is changing and making the target outdated.  But, I'd like to be able to confirm that and find the specific object that is causing the problem.

Is there a function or a reporter that can show specifics of why a target is outdated?  I think that my ideal case would to find something that shows the digest or other supporting information for why the underlying object caused the target to be outdated.

**Comment by wlandau** [comment]
tar_sitrep() can do this. I hope it helps.

**Reply by billdenney** [reply, marked as answer]
`tar_sitrep()` shows me that `depend = TRUE` for the steps that are rerunning.

After looking a bit more, `tar_visnetwork()` showed me the information I was looking for visually-- that the expected objects were outdated and they were the only dependencies that were outdated.  I then found the documentation of the `_targets/meta/meta` file which indicated that the third field is the hash of the objects which is changing with each run (https://books.ropensci.org/targets-design/data.html#metadata).

I think that I found what I was looking for.
